samples: buck: rework BUCK example into 4 separate ones (#74)
NordicSemiconductor/npmx-zephyr: samples/buck/output_voltage_advanced/README.rst

> .. _buck_sample:

Reviewer: ```suggestion
.. _buck_output_voltage_advanced_sample:
```
The same for other samples.

> BUCK - set advanced output voltage
> ####

Reviewer: ```suggestion
BUCK - set advanced output voltage
##################################
```

.. contents::
   :local:
   :depth: 2

> The BUCK samples demonstrates how to control PMIC BUCKs using npmx drivers.

Reviewer: ```suggestion
The BUCK Output Voltage Advanced sample demonstrates how to control PMIC BUCKs using npmx drivers.
```

@annakielar should it be here with uppercase?

************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM1300 EK.

Overview
********

This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.
The BUCK1 starts from voltage 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`).
The output voltage increases from the selected voltage to the maximum one (3.3 V).

Voltage can be changed only for BUCK1.
BUCK2 voltage is set by default to 3.0 V.

Wiring
******

#. Connect the TWI interface between the chosen DK and the nPM1300 EK as in the following table:

   .. list-table:: nPM1300 EK connections.
     :widths: 25 25 25 25
     :header-rows: 1

     * - nPM1300 EK pins
       - nRF5340 DK pins
       - nRF52840 DK pins
       - nRF52832 DK pins
     * - GPIO0
       - P1.10
       - P1.10
       - P0.24
     * - SDA
       - P1.02
       - P0.26
       - P0.26
     * - SCL
       - P1.03
       - P0.27
       - P0.27
     * - VOUT2 & GND
       - External supply (P21)
       - External supply (P21)
       - External supply (P21)

#. Make the following connections on the nPM1300 EK:

   * Connect a USB power supply to the **J3** connector.

>    * Connect a suitable battery to the **J2** connector.

Reviewer: Battery is not required for this sample
